### ISSUE TITLE
Bad structure

### ISSUE THREAD

**Issue by evi-Genius** [google-research/morph-net#144]
HI, I have pruned the resnet18 with flops regularizer, but the final structure of pruned resnet18 model is strange. The first conv of 7×7 kernel and  64 filter  is pruned to **zero** channels, so that the network can not run. Do you have any idea to deal with this situation?

**Comment by ayp-google** [comment]
Try using a smaller regularization strength.  If the regularization strength is too large, then the model will collapse because the optimizer will value the regularization loss over the training loss.

**Comment by eladeban** [comment]
You could try a smaller regularization strength. 

Zero-ing out an entire layer is an edge case that happens as files are not 'alive' anymore but somehow used by the network (having a small magnitude dose not prevent the network from learning using a filter).

**Comment by abelandcain** [comment]
i tried small regularization strength .now also its zeroing out.i am using resnet 50 with cifar 10 .layers are being zeroed in alive_0 file only.no matter strength is 1e-4 or 1e-6
